winsock2.h includes cause compiler errors.

On attempting to compile network lib you get the following errors:

1>C:\Program Files (x86)\Microsoft Visual Studio 8\VC\PlatformSDK\include\winsock2.h(112) : error C2011: 'fd_set' : 'struct' type redefinition
1>        C:\Program Files (x86)\Microsoft Visual Studio 8\VC\PlatformSDK\include\winsock.h(54) : see declaration of 'fd_set'
1>C:\Program Files (x86)\Microsoft Visual Studio 8\VC\PlatformSDK\include\winsock2.h(147) : warning C4005: 'FD_SET' : macro redefinition
1>        C:\Program Files (x86)\Microsoft Visual Studio 8\VC\PlatformSDK\include\winsock.h(88) : see previous definition of 'FD_SET'
1>C:\Program Files (x86)\Microsoft Visual Studio 8\VC\PlatformSDK\include\winsock2.h(156) : error C2011: 'timeval' : 'struct' type redefinition
1>        C:\Program Files (x86)\Microsoft Visual Studio 8\VC\PlatformSDK\include\winsock.h(97) : see declaration of 'timeval'
1>C:\Program Files (x86)\Microsoft Visual Studio 8\VC\PlatformSDK\include\winsock2.h(212) : error C2011: 'hostent' : 'struct' type redefinition
1>        C:\Program Files (x86)\Microsoft Visual Studio 8\VC\PlatformSDK\include\winsock.h(153) : see declaration of 'hostent'
1>C:\Program Files (x86)\Microsoft Visual Studio 8\VC\PlatformSDK\include\winsock2.h(225) : error C2011: 'netent' : 'struct' type redefinition
1>        C:\Program Files (x86)\Microsoft Visual Studio 8\VC\PlatformSDK\include\winsock.h(166) : see declaration of 'netent'
1>C:\Program Files (x86)\Microsoft Visual Studio 8\VC\PlatformSDK\include\winsock2.h(232) : error C2011: 'servent' : 'struct' type redefinition
1>        C:\Program Files (x86)\Microsoft Visual Studio 8\VC\PlatformSDK\include\winsock.h(173) : see declaration of 'servent'
1>C:\Program Files (x86)\Microsoft Visual Studio 8\VC\PlatformSDK\include\winsock2.h(244) : error C2011: 'protoent' : 'struct' type redefinition
1>        C:\Program Files (x86)\Microsoft Visual Studio 8\VC\PlatformSDK\include\winsock.h(185) : see declaration of 'protoent'
1>C:\Program Files (x86)\Microsoft Visual Studio 8\VC\PlatformSDK\include\winsock2.h(400) : error C2011: 'sockaddr_in' : 'struct' type redefinition
1>        C:\Program Files (x86)\Microsoft Visual Studio 8\VC\PlatformSDK\include\winsock.h(321) : see declaration of 'sockaddr_in'
1>C:\Program Files (x86)\Microsoft Visual Studio 8\VC\PlatformSDK\include\winsock2.h(410) : error C2011: 'WSAData' : 'struct' type redefinition
1>        C:\Program Files (x86)\Microsoft Visual Studio 8\VC\PlatformSDK\include\winsock.h(331) : see declaration of 'WSAData'
1>C:\Program Files (x86)\Microsoft Visual Studio 8\VC\PlatformSDK\include\winsock2.h(470) : warning C4005: 'SO_DONTLINGER' : macro redefinition
1>        C:\Program Files (x86)\Microsoft Visual Studio 8\VC\PlatformSDK\include\winsock.h(411) : see previous definition of 'SO_DONTLINGER'
1>C:\Program Files (x86)\Microsoft Visual Studio 8\VC\PlatformSDK\include\winsock2.h(527) : warning C4005: 'AF_IPX' : macro redefinition
1>        C:\Program Files (x86)\Microsoft Visual Studio 8\VC\PlatformSDK\include\winsock.h(469) : see previous definition of 'AF_IPX'
1>C:\Program Files (x86)\Microsoft Visual Studio 8\VC\PlatformSDK\include\winsock2.h(555) : warning C4005: 'AF_MAX' : macro redefinition
1>        C:\Program Files (x86)\Microsoft Visual Studio 8\VC\PlatformSDK\include\winsock.h(488) : see previous definition of 'AF_MAX'
1>C:\Program Files (x86)\Microsoft Visual Studio 8\VC\PlatformSDK\include\winsock2.h(561) : error C2011: 'sockaddr' : 'struct' type redefinition
1>        C:\Program Files (x86)\Microsoft Visual Studio 8\VC\PlatformSDK\include\winsock.h(494) : see declaration of 'sockaddr'
1>C:\Program Files (x86)\Microsoft Visual Studio 8\VC\PlatformSDK\include\winsock2.h(601) : error C2011: 'sockproto' : 'struct' type redefinition
1>        C:\Program Files (x86)\Microsoft Visual Studio 8\VC\PlatformSDK\include\winsock.h(503) : see declaration of 'sockproto'
1>C:\Program Files (x86)\Microsoft Visual Studio 8\VC\PlatformSDK\include\winsock2.h(640) : error C2011: 'linger' : 'struct' type redefinition
1>        C:\Program Files (x86)\Microsoft Visual Studio 8\VC\PlatformSDK\include\winsock.h(540) : see declaration of 'linger'
1>C:\Program Files (x86)\Microsoft Visual Studio 8\VC\PlatformSDK\include\winsock2.h(653) : warning C4005: 'SOMAXCONN' : macro redefinition
1>        C:\Program Files (x86)\Microsoft Visual Studio 8\VC\PlatformSDK\include\winsock.h(553) : see previous definition of 'SOMAXCONN'
1>C:\Program Files (x86)\Microsoft Visual Studio 8\VC\PlatformSDK\include\winsock2.h(679) : warning C4005: 'FD_READ' : macro redefinition
1>        C:\Program Files (x86)\Microsoft Visual Studio 8\VC\PlatformSDK\include\winsock.h(571) : see previous definition of 'FD_READ'
1>C:\Program Files (x86)\Microsoft Visual Studio 8\VC\PlatformSDK\include\winsock2.h(682) : warning C4005: 'FD_WRITE' : macro redefinition
1>        C:\Program Files (x86)\Microsoft Visual Studio 8\VC\PlatformSDK\include\winsock.h(572) : see previous definition of 'FD_WRITE'
1>C:\Program Files (x86)\Microsoft Visual Studio 8\VC\PlatformSDK\include\winsock2.h(685) : warning C4005: 'FD_OOB' : macro redefinition
1>        C:\Program Files (x86)\Microsoft Visual Studio 8\VC\PlatformSDK\include\winsock.h(573) : see previous definition of 'FD_OOB'
1>C:\Program Files (x86)\Microsoft Visual Studio 8\VC\PlatformSDK\include\winsock2.h(688) : warning C4005: 'FD_ACCEPT' : macro redefinition
1>        C:\Program Files (x86)\Microsoft Visual Studio 8\VC\PlatformSDK\include\winsock.h(574) : see previous definition of 'FD_ACCEPT'
1>C:\Program Files (x86)\Microsoft Visual Studio 8\VC\PlatformSDK\include\winsock2.h(691) : warning C4005: 'FD_CONNECT' : macro redefinition
1>        C:\Program Files (x86)\Microsoft Visual Studio 8\VC\PlatformSDK\include\winsock.h(575) : see previous definition of 'FD_CONNECT'
1>C:\Program Files (x86)\Microsoft Visual Studio 8\VC\PlatformSDK\include\winsock2.h(694) : warning C4005: 'FD_CLOSE' : macro redefinition
1>        C:\Program Files (x86)\Microsoft Visual Studio 8\VC\PlatformSDK\include\winsock.h(576) : see previous definition of 'FD_CLOSE'
1>C:\Program Files (x86)\Microsoft Visual Studio 8\VC\PlatformSDK\include\winsock2.h(1626) : error C2375: 'accept' : redefinition; different linkage
1>        C:\Program Files (x86)\Microsoft Visual Studio 8\VC\PlatformSDK\include\winsock.h(744) : see declaration of 'accept'
1>C:\Program Files (x86)\Microsoft Visual Studio 8\VC\PlatformSDK\include\winsock2.h(1647) : error C2375: 'bind' : redefinition; different linkage
1>        C:\Program Files (x86)\Microsoft Visual Studio 8\VC\PlatformSDK\include\winsock.h(749) : see declaration of 'bind'
1>C:\Program Files (x86)\Microsoft Visual Studio 8\VC\PlatformSDK\include\winsock2.h(1666) : error C2375: 'closesocket' : redefinition; different linkage

as well as many more, however they are telling the same idea:
the includes of winsock2.h are causing redefinitions of certain structs and macros

--- a/Client/App/include/util/Profiling.h
+++ b/Client/App/include/util/Profiling.h
@@ -1,8 +1,7 @@
 #pragma once
-#include <windows.h>
+#include <G3D/Vector3.h>
 #include <boost/noncopyable.hpp>
 #include <string>
-#include <g3d/g3dmath.h>
 
 namespace RBX
 {

--- a/Client/Network/Client.cpp
+++ b/Client/Network/Client.cpp
@@ -1,7 +1,8 @@
-#include "util/standardout.h"
+#include <RakNetTypes.h>
+#include <RakPeer.h>
 #include "API.h"
 #include "Client.h"
-#include <RakNetTypes.h>
+#include "util/standardout.h"
 
 const char* Exposer::IDTOString(int id)
 {

--- a/Client/Network/Client.h
+++ b/Client/Network/Client.h
@@ -1,8 +1,7 @@
 #pragma once
-#include <winsock2.h>
-#include "reflection/signal.h"
-#include "v8xml/XmlElement.h"
+#include <PacketLogger.h>
 #include "Replicator.h"
+#include "reflection/signal.h"
 #include <boost/shared_ptr.hpp>
 #include <string>
 
@@ -13,6 +12,8 @@ public:
 
 	static const char* IDTOString(int id);
 };
+
+class XmlElement;
 
 namespace RBX
 {

--- a/Client/Network/Player.cpp
+++ b/Client/Network/Player.cpp
@@ -1,5 +1,5 @@
-#include "Player.h"
 #include "Client.h"
+#include "Player.h"
 #include "v8datamodel/TimerService.h"
 
 namespace RBX

--- a/Client/Network/Players.cpp
+++ b/Client/Network/Players.cpp
@@ -1,5 +1,7 @@
-#include "Players.h"
+#include <RakPeer.h>
+#include <BitStream.h>
 #include "Client.h"
+#include "Players.h"
 #include "Streaming.h"
 
 template<class Class>

--- a/Client/Network/Replicator.cpp
+++ b/Client/Network/Replicator.cpp
@@ -1,3 +1,5 @@
+#include <RakPeer.h>
+#include <PacketLogger.h>
 #include "Replicator.h"
 #include <g3d/system.h>
 

--- a/Client/Network/Replicator.h
+++ b/Client/Network/Replicator.h
@@ -1,15 +1,14 @@
 #pragma once
-#include <winsock2.h>
-#include "v8tree/Instance.h"
-#include "v8tree/Service.h"
+#include <PluginInterface.h>
+#include "util/RunStateOwner.h"
 #include "util/Events.h"
 #include "util/Profiling.h"
-#include "util/RunStateOwner.h"
-#include <PacketLogger.h>
-#include <PluginInterface.h>
-#include <RakPeer.h>
-#include <RakPeerInterface.h>
 #include <boost/scoped_ptr.hpp>
+
+class RakPeer;
+class RakPeerInterface;
+
+class PacketLogger;
 
 namespace RBX
 {

--- a/Client/Network/Streaming.cpp
+++ b/Client/Network/Streaming.cpp
@@ -1,6 +1,6 @@
-#include "Streaming.h"
 #include <BitStream.h>
 #include <StringCompressor.h>
+#include "Streaming.h"
 #include <algorithm>
 
 namespace RBX

--- a/Client/Network/Streaming.h
+++ b/Client/Network/Streaming.h
@@ -4,12 +4,16 @@
 #include "util/Name.h"
 #include "util/Guid.h"
 #include "reflection/property.h"
-#include <BitStream.h>
 #include <g3d/CoordinateFrame.h>
 #include <boost/shared_ptr.hpp>
 #include <boost/noncopyable.hpp>
 #include <map>
 #include <string>
+
+namespace RakNet
+{
+	class BitStream;
+}
 
 namespace RBX
 {

--- a/Client/Network/include/Network/Player.h
+++ b/Client/Network/include/Network/Player.h
@@ -1,5 +1,4 @@
 #pragma once
-#include <winsock2.h>
 #include "Network/Players.h"
 #include "security/SecurityContext.h"
 #include "humanoid/Humanoid.h"

